src/bentoml/_internal/io_descriptors/numpy.py

```diff
@@ -471,6 +471,11 @@ def _from_sample(self, sample: ext.NpNDArray | t.Sequence[t.Any]) -> ext.NpNDArr
            @svc.api(input=input_spec, output=NumpyNdarray())
            async def predict(input: NDArray[np.int16]) -> NDArray[Any]:
                return await runner.async_run(input)
+
+        Raises:
+            :class:`BentoMLException`: If given sample is a type ``numpy.generic``. This exception
+                                       will also be raised if we failed to create a ``np.ndarray``
+                                       from given sample.
```

Comment:
If we're going to document this we should probably make it a better exception.

Reply:
got it. Maybe I can put the exception PR up 😄